pandas/tests/io/test_gcs.py

```diff
@@ -0,0 +1,25 @@
+from pandas import DataFrame, read_csv
+from pandas.compat import BytesIO
+from pandas.io.common import is_gcs_url
+
+
+def test_is_gcs_url():
+    assert is_gcs_url("gcs://pandas/somethingelse.com")
+    assert is_gcs_url("gs://pandas/somethingelse.com")
+    assert not is_gcs_url("s3://pandas/somethingelse.com")
+
+
+def test_read_csv_gcs():
+    try:
+        from unittest.mock import patch
+    except ImportError:
+        from mock import patch
+
+    with patch('gcsfs.GCSFileSystem') as MockFileSystem:
+        instance = MockFileSystem.return_value
+        instance.open.return_value = BytesIO(b'a,b\n1,2\n3,4')
+        df = read_csv('gs://test/test.csv')
```

Comment:
Mock let's you assert that certain functions are called, right?

I guess, maybe we could have a second test that ensures `pandas.io.gcs.get_filepath_or_buffer` is called for `gs://` type `filepath_or_buffer`?